test/alu_8bit_adc_test.c

```diff
@@ -0,0 +1,248 @@
+#include <criterion/criterion.h>
+#include <criterion/new/assert.h>
+#include <criterion/parameterized.h>
+
+#include "fixtures/cpu_mmu.h"
+
+typedef struct TestParams {
+    uint8_t opcode;
+    uint8_t lhs;
+    uint8_t rhs;
+    size_t rhs_dreg_offset;
+    size_t rhs_reg_offset;
+    uint8_t expected;
+    int address_increment;
+    uint8_t F;
+} TestParams;
+
+typedef struct SpecialTestParams {
+    uint8_t opcode;
+    uint8_t lhs;
+    uint8_t rhs;
+    bool uses_borrow;
+    uint8_t expected;
+    uint8_t F;
+    bool is_HL;
+} SpecialTestParams;
+
+static void run_test(TestParams const *const params) {
+    // setup cpu
+    uint16_t address = (random() % (MEM_SIZE - ROM_LIMIT)) + ROM_LIMIT;
+    cpu.PC           = address;
+    CPU_REG_A        = params->lhs;
+
+    // write rhs to desired register
+    char *target_reg       = ((char *) &cpu) + params->rhs_dreg_offset;
+    uint8_t *target_nibble = (uint8_t *) (target_reg + params->rhs_reg_offset);
+    *(target_nibble)       = params->rhs;
```

Comment:
I think that I will move this bit of code into common together with the code in `lds_test`

Reply:
Nvm, it's already there, just changing this